code/__HELPERS/_lists.dm

```diff
@@ -11,20 +11,24 @@
 
 #define LAZYINITLIST(L) if (!L) L = list()
 #define UNSETEMPTY(L) if (L && !length(L)) L = null
-#define LAZYCOPY(L) (L ? L.Copy() : list() )
+#define LAZYLISTDUPLICATE(L) (L ? L.Copy() : null )
 #define LAZYREMOVE(L, I) if(L) { L -= I; if(!length(L)) { L = null; } }
 #define LAZYADD(L, I) if(!L) { L = list(); } L += I;
 #define LAZYOR(L, I) if(!L) { L = list(); } L |= I;
 #define LAZYFIND(L, V) (L ? L.Find(V) : 0)
 #define LAZYACCESS(L, I) (L ? (isnum(I) ? (I > 0 && I <= length(L) ? L[I] : null) : L[I]) : null)
 #define LAZYSET(L, K, V) if(!L) { L = list(); } L[K] = V;
 #define LAZYLEN(L) length(L)
-#define LAZYCLEARLIST(L) if(L) L.Cut()
-#define SANITIZE_LIST(L) ( islist(L) ? L : list() )
-#define reverseList(L) reverseRange(L.Copy())
+#define LAZYNULL(L) L = null
 #define LAZYADDASSOC(L, K, V) if(!L) { L = list(); } L[K] += list(V);
 #define LAZYREMOVEASSOC(L, K, V) if(L) { if(L[K]) { L[K] -= V; if(!length(L[K])) L -= K; } if(!length(L)) L = null; }
 #define LAZYACCESSASSOC(L, I, K) L ? L[I] ? L[I][K] ? L[I][K] : null : null : null
+#define QDEL_LAZYLIST(L) for(var/I in L) qdel(I); L = null;
+//Not to be used with actual lazy lists, as these don't null it on clearing.
+#define LAZYCOPY(L) (L ? L.Copy() : list() ) //Use LAZYLISTDUPLICATE instead - legacy code
```

Comment:
This isn't necessarily a bad thing.

I used it for instance with `get_special_statuses`:

```dm
/datum/mind/proc/get_special_statuses()
	var/list/result = LAZYCOPY(special_statuses)
	if(!current)
		result += "<span class='bad'>No body!</span>"
	if(current && HAS_TRAIT(current, TRAIT_MINDSHIELD))
		result += "<span class='good'>Mindshielded</span>"
	//Move these to mob
	if(iscyborg(current))
		var/mob/living/silicon/robot/robot = current
		if (robot.emagged)
			result += "<span class='bad'>Emagged</span>"
	return result.Join(" | ")
```

Doing `LAZYADD` every time here would've been a bit clunky.

Reply:
See above

Reply:
I edited the comments though so that they're a bit more in line with what you're saying